app/webpacker/components/CompetitionsOverview/useDelegatesData.js

```diff
@@ -1,36 +1,18 @@
-import { useEffect } from 'react';
-import { useInfiniteQuery } from '@tanstack/react-query';
+import { useQuery } from '@tanstack/react-query';
 
-import { apiV0Urls, WCA_API_PAGINATION } from '../../lib/requests/routes.js.erb';
+import { apiV0Urls } from '../../lib/requests/routes.js.erb';
 import { fetchJsonOrError } from '../../lib/requests/fetchWithAuthenticityToken';
 
 const useDelegatesData = () => {
   const {
     data,
-    fetchNextPage,
-    hasNextPage,
-  } = useInfiniteQuery({
-    queryKey: ['delegates'],
-    queryFn: ({ pageParam = 1 }) => fetchJsonOrError(`${apiV0Urls.delegates.list}?page=${pageParam}`),
-    getNextPageParam: (previousPage, allPages) => {
-      // Continue until less than a full page of data is fetched,
-      // which indicates the very last page.
-      if (previousPage.data.length < WCA_API_PAGINATION) {
-        return undefined;
-      }
-      return allPages.length + 1;
-    },
+    isPending,
+  } = useQuery({
+    queryKey: ['delegates-index'],
+    queryFn: () => fetchJsonOrError(apiV0Urls.delegates.searchIndex),
   });
 
-  useEffect(() => {
-    if (hasNextPage) {
-      fetchNextPage();
-    }
-  }, [data, hasNextPage, fetchNextPage]);
-
-  const delegatesData = data?.pages.flatMap((page) => page.data);
-
-  return { delegatesLoading: hasNextPage, delegatesData };
+  return { delegatesLoading: isPending, delegatesData: data?.data };
```

Comment:
`data.data` looks a little weird, how come the data isn't at the top level like it used to be?

Reply:
This was actually a problem in the previous, paginated implementation as well. Quote: (see left side of diff)

```
const delegatesData = data?.pages.flatMap((page) => page.data);
                      ^^^^                               ^^^^
```

The problem here is that our `fetchJsonOrError` was originally conceived as a standalone solution, independent of Tanstack-query. That's why it returns a custom-build object with properties `data` and `headers`.

Now Tanstack-query is a fully-fledged solution that offers its own return value to access headers, and `data` just happens to be the name of whatever `fetchJsonOrError` returns.

An immediate solution here would be to adjust the `queryFn` to resolve as `fetchJsonOrError(...).then((resp) => resp.data)`, but I'd rather wait for a future "replace `useLoadedData` with Tanstack-Query" PR where we think this through properly, and accept the mediocre `data?.data` solution for the time being.